keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Beta.java

```diff
@@ -7,66 +7,99 @@
 import io.improbable.keanu.distributions.ContinuousDistribution;
 import io.improbable.keanu.distributions.hyperparam.Diffs;
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.LogProbGraph;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.PlaceHolderDoubleVertex;
 
 public class Beta implements ContinuousDistribution {
 
-    private final DoubleTensor alpha;
-    private final DoubleTensor beta;
-    private final DoubleTensor xMin;
-    private final DoubleTensor xMax;
+    private final DoubleVertex x;
+    private final DoubleVertex alpha;
+    private final DoubleVertex beta;
+    private final DoubleVertex xMin;
+    private final DoubleVertex xMax;
```

Comment:
I feel uneasy about introducing `Vertex` classes into the `Distribution` classes. The way it is now feels neat, that `Distributions` deal with `Tensor`s and need know nothing about `Vertex`es.

Reply:
2 possible ways to change it:
1. Make `logProbGraph` a method on `BetaVertex` rather than on `Beta`
2. Build your log prob graph out of something other than `Vertex`es (as per my comment later on in `LogProbGraph.java`)

Reply:
DoubleTensors is a computation concern where as Vertex is a "problem description" (maybe need a better name for this) concern. The question is where does the Beta distribution live? The BetaVertex is clearly on the "problem description" side and it will depend on some distribution functionality, which will now require a graph. If the distribution supplies a graph then I think it stands to reason that it should also live on the description side. This might mean that we can just move the graph based code from the distribution into the random variable vertex and then drop the distribution classes. So where does the DoubleTensor computation functionality now live? Well, if our inference/computation requires a compiled graph then we would only ever need the DoubleTensor functions in our compiled graph, which may be a compiled Tensorflow graph or something of our own until we can get full functionality out of Tensorflow. If its TF then we don't need the DoubleTensor code at all and if its something of our own then I'd say we'd want to generate the DoubleTensor code based on the same graph that we would use for the TF conversion.

Reply:
Yes, I would say that the distribution classes are a computation concern.

Our discussion below offers a 3rd option, which is that we redefine `Vertex` so that it's something more general with a smaller API. Then I would be comfortable mixing `Vertex` with `Distribution`.

Reply:
`Distribution` as a DoubleTensor only class will go away when everything is a graph. Until everything is a graph, we need it to still provide the old `DoubleTensor` API. When everything is a graph, the issue won't be wether `Distribution` should be coupled to `Vertex` because there won't be any non-`Vertex` functionality left for the `Distribution`.